ddl/table.go

```diff
@@ -621,34 +632,21 @@ func onTruncateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ erro
 		}
 	}
 
-	is := d.infoCache.GetLatest()
-
-	bundles := make([]*placement.Bundle, 0, len(oldPartitionIDs)+1)
-	if oldBundle, ok := is.BundleByName(placement.GroupID(tableID)); ok {
-		bundles = append(bundles, oldBundle.Clone().Reset(placement.RuleIndexTable, []int64{newTableID}))
-	}
-
 	if pi := tblInfo.GetPartitionInfo(); pi != nil {
 		oldIDs := make([]int64, 0, len(oldPartitionIDs))
 		newIDs := make([]int64, 0, len(oldPartitionIDs))
 		newDefs := pi.Definitions
 		for i := range oldPartitionIDs {
-			newID := newDefs[i].ID
-			if oldBundle, ok := is.BundleByName(placement.GroupID(oldPartitionIDs[i])); ok && !oldBundle.IsEmpty() {
+			newDef := &newDefs[i]
+			newID := newDef.ID
+			if newDef.PlacementPolicyRef != nil || newDef.DirectPlacementOpts != nil {
```

Comment:
from the semantic, using oldPartitionIDs[i].PlacementPolicyRef != nil ... will be more suitable

Reply:
I think both is fine because they are the same for placement settings. Here we use newDef is that we cannot get the "oldDef" because tableInfo is already modified to a new one.